squad/frontend/templates/squad/failures.jinja2

```diff
@@ -0,0 +1,80 @@
+{% extends "squad/base.jinja2" %}
+
+{% block content %}
+
+<div ng-app='Build'>
+{% include "squad/build-nav.jinja2" %}
+</div>
+
+<h2>{{ _('All test failures') }}</h2>
+
+<div>
+    <div class='row row-bordered'>
+        <div class='col-md-12 col-sm-12 filter'>
+          <a id="searchLink"><button type='button' class='btn btn-primary fa fa-search'></button></a>
+          <input name='search' id='search' type='text' placeholder='{{ _('Filter results ...') }}' value='{{search}}' />
+        </div>
+    </div>
+
+    <table class='test-results'>
+        <thead>
+            <th>{{ _('Test') }}</th>
+            {% for e in environments %}
+            <th>{{ e.slug }}</th>
+            {% endfor %}
+        </thead>
+        {% if not ufailures %}
+        <tr>
+            <td colspan="{{environments|length|add(1)}}" class="alert alert-warning">
+                <em>{{ _('This build has no failures yet.') }}</em>
+            </td>
+        </tr>
+        {% else %}
+            {% for u in ufailures %}
+              <tr>
+                <td>{{ u.full_name }}</td>
+                {% for e in environments %}
+                {% if e.slug in rows and u.full_name in rows[e.slug]: %}
+                  <td class="{{ rows[e.slug][u.full_name].status }}">
+                    {{ rows[e.slug][u.full_name].status }}
+                    <span data-toggle="tooltip" title="{{ _('Click to display confidence info') }}">
+                      <button class="fa fa-info-circle popover-regressions-fixes"></button>
+                      <span title="{{ _('Confidence') }}" class="hidden">
+                        Score: {{ rows[e.slug][u.full_name].confidence.score  }}
+                      </span>
```

Comment:
it's a bit funny for you to go through all this trouble to then hide the confidence score in a tooltip. I know it's tricky given the volume of information in this table, but maybe it would make sense to use some color coding, e.g. a low confidence score having very light red background, and a high confidence score having a strong red background, while keeping?

you can save some space by not displaying the "fail 🛈" string in the cell body: since this page only lists failures, each cell is either empty meaning "not a failure", or non-empty, meaning a failure. You could even display the confidence score directly in the cell contents.

Reply:
Very true. Some things to consider that are planned that might explain a few concerns.

Right now it's about failures and how confident we're about them being actual failures, but the plan is to use this same method to discover expected failures that are not failures anymore. The problem we have today is that the list of xfails is passed on from generation to generation (of stable branches, that is), so they are rarely reviewed. We'd like to find out when that's changed.

The other is that we want to reuse this confidence index (for failures, for passes) through an API. I see these changes are in the frontend/ directory, but am hoping that this code can be reused.

I'm not sure about the hidden score, though. Justin and I discussed this a bit and I wanted to have the cell information simplified to just the score (not even the label), and have any extra information provided via a tooltip, but the score definitely needs to be shown on the cell.

Reply:
Fixed.

I thought about using a color scheme, but @mrchapp was not thrilled with the idea. For now, just display the score and have the other information available via tooltip.